flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/functions/CollectionFunctionsITCase.java

```diff
@@ -0,0 +1,261 @@
+                        .testSqlError(
+                                "MAP_FROM_ARRAYS(array[1, 2, 3], array[1, 2])",
+                                "Invalid function MAP_FROM_ARRAYS call:\n"
+                                        + "The length of the keys array 3 is not equal to the length of the values array 2"),
+                TestSpec.forFunction(BuiltInFunctionDefinitions.MAP_FROM_ARRAYS)
+                        .onFieldsWithData(1, "one", 2, "two")
+                        .andDataTypes(
+                                DataTypes.INT(),
+                                DataTypes.STRING(),
+                                DataTypes.INT(),
+                                DataTypes.STRING())
+                        .testTableApiResult(
+                                call(
+                                        "cardinality",
+                                        mapFromArrays(
+                                                        array($("f1"), $("f3")),
+                                                        array($("f0"), $("f2")))
+                                                .mapValues()),
+                                2,
+                                DataTypes.INT())
+                        .testResult(
+                                mapFromArrays(array($("f0"), $("f2")), array($("f1"), $("f3"))),
+                                "MAP_FROM_ARRAYS(array[f0, f2], array[f1, f3])",
+                                ImmutableMap.of(1, "one", 2, "two"),
+                                DataTypes.MAP(DataTypes.INT(), DataTypes.STRING()))
+                        .testTableApiResult(
+                                mapFromArrays(array($("f2"), $("f0")), array($("f3"), $("f1"))),
+                                ImmutableMap.of(1, "one", 2, "two"),
+                                DataTypes.MAP(DataTypes.INT(), DataTypes.STRING()))
+                        .testResult(
+                                mapFromArrays(
+                                        array(map($("f0"), $("f2"))), array(map($("f1"), $("f3")))),
+                                "MAP_FROM_ARRAYS(array[map[f0, f2]], array[map[f1, f3]])",
+                                ImmutableMap.of(
+                                        ImmutableMap.of(1, 2), ImmutableMap.of("one", "two")),
+                                DataTypes.MAP(
+                                        DataTypes.MAP(DataTypes.INT(), DataTypes.INT()).notNull(),
+                                        DataTypes.MAP(DataTypes.STRING(), DataTypes.STRING())
+                                                .notNull()))
+                        .testResult(
+                                mapFromArrays(array($("f0")), array(map($("f1"), $("f3")))),
+                                "MAP_FROM_ARRAYS(array[f0], array[map[f1, f3]])",
+                                ImmutableMap.of(1, ImmutableMap.of("one", "two")),
```

Comment:
we try to avoid Guava in our code base, you can create a helper method to create the map but in this case a `Collections.singletonMap` should do the job.

Reply:
ok, I didn't know that it is better to avoid Guava => removed it